### PR TITLE
fix/overflow-in-sci-clubs-card

### PR DESCRIPTION
**Pull request by tomasz-trela**

### CODE REVIEW

**Review by simon-the-shark — CHANGES_REQUESTED** [review]
sorry to say this but I don't think this is the way we want to handle this

Zrzut ekranu 2024-09-26 o 00 33 38

5 lines look much better but it has problems in few edge cases I agree. We can discuss how we want to handle that

**Comment by TTomaszPWR** [comment]
> sorry to say this but I don't think this is the way we want to handle this
> 
> Zrzut ekranu 2024-09-26 o 00 33 38
> 
> 5 lines look much better but it has problems in few edge cases I agree. We can discuss how we want to handle that

Hmm, we can limit lines for science club name. I think it is a good idea. What do you think @simon-the-shark ?

**Comment by mikolaj-jalocha** [comment]
> > sorry to say this but I don't think this is the way we want to handle this
> > Zrzut ekranu 2024-09-26 o 00 33 38
> > 5 lines look much better but it has problems in few edge cases I agree. We can discuss how we want to handle that
> 
> Hmm, we can limit lines for science club name. I think it is a good idea. What do you think @simon-the-shark ?

I'm not sure what's the exact problem but it's long names of organizations I suppose. I'd think about some kind of scalling long names. Imo they should be visible. Similar thing was done at home's screen section regarding sci clubs. You can check the solution there, maybe it'd suit for this as well.

**Comment by TTomaszPWR** [comment]
> > > sorry to say this but I don't think this is the way we want to handle this
> > > Zrzut ekranu 2024-09-26 o 00 33 38
> > > 5 lines look much better but it has problems in few edge cases I agree. We can discuss how we want to handle that
> > 
> > 
> > Hmm, we can limit lines for science club name. I think it is a good idea. What do you think @simon-the-shark ?
> 
> I'm not sure what's the exact problem but it's long names of organizations I suppose. I'd think about some kind of scalling long names. Imo they should be visible. Similar thing was done at home's screen section regarding sci clubs. You can check the solution there, maybe it'd suit for this as well.

That's sounds better. I'll do it.

**Comment by TTomaszPWR** [comment]
After many attempts to auto-size text. I've decided to use pub dev library.

File: lib/features/science_clubs_view/widgets/ensure_visible_tags.dart [hunk 1]
```diff
@@ -46,10 +47,10 @@ class EnsureVisibleTags extends DualTextMaxLines {
         return Column(
           crossAxisAlignment: CrossAxisAlignment.stretch,
           children: [
-            RichText(
+            AutoSizeText.rich(
```
Comment (simon-the-shark) [inline lib/features/science_clubs_view/widgets/ensure_visible_tags.dart:50]: its' actually a real funny combination that we do calculate the max lines with our logic and then apply scaling down but the results looks pretty ok

**Review by simon-the-shark — APPROVED** [review]

**Comment by simon-the-shark** [comment]
I'll maybe try to limit department to 1 line too - maybe the tags will be better looking. But merging this for now

**Pull request merged**